Sources/MongoSwift/MongoClient.swift

```diff
@@ -420,11 +420,8 @@ public class MongoClient {
                                   options: ChangeStreamOptions?  =  nil,
                                   session: ClientSession? = nil,
                                   withEventType: T.Type) throws -> ChangeStream<T> {
-        let pipeline: Document = ["pipeline": pipeline]
-        let opts = try encodeOptions(options: options, session: session)
-        return try ChangeStream<T>(options: options, client: self, decoder: self.decoder, session: session) { conn in
-            mongoc_client_watch(conn.clientHandle, pipeline._bson, opts?._bson)
-        }
+        let operation = try WatchOperation<T, T>(target: .client(self), pipeline: pipeline, options: options)
```

Comment:
Let's update the generic argument to `watch` to disambiguate from the collection's parameter, maybe `EventType` would be better? Then we can use `WatchOperation<CollectionType, EventType>`. This also applies to other `watch` functions.